assembly/src/ast/mod.rs

```diff
@@ -336,6 +338,23 @@ impl ProgramAst {
     pub fn clear_imports(&mut self) {
         self.import_info = None;
     }
+
+    // WRITE TO FILE
+    // --------------------------------------------------------------------------------------------
+    #[cfg(feature = "std")]
+    /// Writes ProgramAst to provided file path
+    pub fn write_to_file<P>(&self, dir_path: P) -> io::Result<()>
+    where
```

Comment:
A couple of minor nits:

1. Let's put the attribute below the comment.
2. Let's rename `dir_path` to `file_path`.